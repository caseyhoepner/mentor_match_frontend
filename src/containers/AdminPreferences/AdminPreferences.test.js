import React from 'react';
import { shallow } from 'enzyme';
import { AdminPreferences, mapDispatchToProps } from './';

describe('AdminPreferences', () => {
  let wrapper;
  let mockFunc = jest.fn()
  
  beforeEach(() => {
    wrapper = shallow(<AdminPreferences
      changeMentorFilters={mockFunc}
    />)
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('toggleVeteran function', () => {
    it('should toggle the veteran boolean in state', async () => {
      expect(wrapper.state().veteran).toEqual(false)
      await wrapper.instance().toggleVeteran()
      expect(wrapper.state().veteran).toEqual(true)
    });

    it('should fire changeMentorFilters with altered state as params', async () => {
      let mockAlteredState = {
        veteran: true,
        parent: false,
        lgbtq: false,
        female: false,
        male: false,
        frontEnd: false,
        backEnd: false
      }

      await wrapper.instance().toggleVeteran()
      expect(mockFunc).toHaveBeenCalledWith(mockAlteredState)
    });
  });

  describe('toggleLgbtq function', () => {
    it('should toggle the lgtbq boolean in state', async () => {
      expect(wrapper.state().lgbtq).toEqual(false)
      await wrapper.instance().toggleLgbtq()
      expect(wrapper.state().lgbtq).toEqual(true)
    });

    it('should fire changeMentorFilters with altered state as params', async () => {
      let mockAlteredState = {
        veteran: false,
        parent: false,
        lgbtq: true,
        female: false,
        male: false,
        frontEnd: false,
        backEnd: false
      }

      await wrapper.instance().toggleLgbtq()
      expect(mockFunc).toHaveBeenCalledWith(mockAlteredState)
    });
  });

  describe('toggleParent function', () => {
    it('should toggle the parent boolean in state', async () => {
      expect(wrapper.state().parent).toEqual(false)
      await wrapper.instance().toggleParent()
      expect(wrapper.state().parent).toEqual(true)
    });

    it('should fire changeMentorFilters with altered state as params', async () => {
      let mockAlteredState = {
        veteran: false,
        parent: true,
        lgbtq: false,
        female: false,
        male: false,
        frontEnd: false,
        backEnd: false
      }

      await wrapper.instance().toggleParent()
      expect(mockFunc).toHaveBeenCalledWith(mockAlteredState)
    });
  });

  describe('toggleFemale function', () => {
    it('should toggle the female boolean in state', async () => {
      expect(wrapper.state().female).toEqual(false)
      await wrapper.instance().toggleFemale()
      expect(wrapper.state().female).toEqual(true)
    });

    it('should fire changeMentorFilters with altered state as params', async () => {
      let mockAlteredState = {
        veteran: false,
        parent: false,
        lgbtq: false,
        female: true,
        male: false,
        frontEnd: false,
        backEnd: false
      }

      await wrapper.instance().toggleFemale()
      expect(mockFunc).toHaveBeenCalledWith(mockAlteredState)
    });
  });

  describe('toggleMale function', () => {
    it('should toggle the male boolean in state', async () => {
      expect(wrapper.state().male).toEqual(false)
      await wrapper.instance().toggleMale()
      expect(wrapper.state().male).toEqual(true)
    });

    it('should fire changeMentorFilters with altered state as params', async () => {
      let mockAlteredState = {
        veteran: false,
        parent: false,
        lgbtq: false,
        female: false,
        male: true,
        frontEnd: false,
        backEnd: false
      }

      await wrapper.instance().toggleMale()
      expect(mockFunc).toHaveBeenCalledWith(mockAlteredState)
    });
  });

  describe('toggleFE function', () => {
    it('should toggle the frontEnd boolean in state', async () => {
      expect(wrapper.state().frontEnd).toEqual(false)
      await wrapper.instance().toggleFE()
      expect(wrapper.state().frontEnd).toEqual(true)
    });

    it('should fire changeMentorFilters with altered state as params', async () => {
      let mockAlteredState = {
        veteran: false,
        parent: false,
        lgbtq: false,
        female: false,
        male: false,
        frontEnd: true,
        backEnd: false
      }

      await wrapper.instance().toggleFE()
      expect(mockFunc).toHaveBeenCalledWith(mockAlteredState)
    });
  });

  describe('toggleBE function', () => {
    it('should toggle the backEnd boolean in state', async () => {
      expect(wrapper.state().backEnd).toEqual(false)
      await wrapper.instance().toggleBE()
      expect(wrapper.state().backEnd).toEqual(true)
    });

    it('should fire changeMentorFilters with altered state as params', async () => {
      let mockAlteredState = {
        veteran: false,
        parent: false,
        lgbtq: false,
        female: false,
        male: false,
        frontEnd: false,
        backEnd: true
      }

      await wrapper.instance().toggleBE()
      expect(mockFunc).toHaveBeenCalledWith(mockAlteredState)
    });
  });
});

describe('mapDispatchToProps function', () => {
  let mockDispatch = jest.fn();
  let mappedProps;

  beforeEach(() => {
    mappedProps = mapDispatchToProps(mockDispatch)
  });

  it('should fire dispatch when changeMentorFilters is called', () => {
    mappedProps.changeMentorFilters()
    expect(mockDispatch).toHaveBeenCalled()
  });
});