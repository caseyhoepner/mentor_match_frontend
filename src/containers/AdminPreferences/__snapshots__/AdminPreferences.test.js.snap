// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPreferences matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminPreferences />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="p-toggle-option"
        >
          <p>
            Veteran
          </p>
          <Switch
            className=""
            enabled={true}
            name="veteran"
            on={false}
            onClick={[Function]}
            onDisabledClick={[Function]}
          />
        </div>,
        <div
          className="p-toggle-option"
        >
          <p>
            Parent
          </p>
          <Switch
            className=""
            enabled={true}
            name="parent"
            on={false}
            onClick={[Function]}
            onDisabledClick={[Function]}
          />
        </div>,
        <div
          className="p-toggle-option"
        >
          <p>
            LGBTQIA+
          </p>
          <Switch
            className=""
            enabled={true}
            name="lgbtq"
            on={false}
            onClick={[Function]}
            onDisabledClick={[Function]}
          />
        </div>,
        <div
          className="p-toggle-option"
        >
          <p>
            Female-Identifying
          </p>
          <Switch
            className=""
            enabled={true}
            name="female"
            on={false}
            onClick={[Function]}
            onDisabledClick={[Function]}
          />
        </div>,
        <div
          className="p-toggle-option"
        >
          <p>
            Male-Identifying
          </p>
          <Switch
            className=""
            enabled={true}
            name="male"
            on={false}
            onClick={[Function]}
            onDisabledClick={[Function]}
          />
        </div>,
        <div
          className="p-toggle-option"
        >
          <p>
            Front-End
          </p>
          <Switch
            className=""
            enabled={true}
            name="frontEnd"
            on={false}
            onClick={[Function]}
            onDisabledClick={[Function]}
          />
        </div>,
        <div
          className="p-toggle-option"
        >
          <p>
            Back-End
          </p>
          <Switch
            className=""
            enabled={true}
            name="backEnd"
            on={false}
            onClick={[Function]}
            onDisabledClick={[Function]}
          />
        </div>,
      ],
      "className": "hide",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Veteran
            </p>,
            <Switch
              className=""
              enabled={true}
              name="veteran"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />,
          ],
          "className": "p-toggle-option",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Veteran",
            },
            "ref": null,
            "rendered": "Veteran",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "enabled": true,
              "name": "veteran",
              "on": false,
              "onClick": [Function],
              "onDisabledClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Parent
            </p>,
            <Switch
              className=""
              enabled={true}
              name="parent"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />,
          ],
          "className": "p-toggle-option",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Parent",
            },
            "ref": null,
            "rendered": "Parent",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "enabled": true,
              "name": "parent",
              "on": false,
              "onClick": [Function],
              "onDisabledClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              LGBTQIA+
            </p>,
            <Switch
              className=""
              enabled={true}
              name="lgbtq"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />,
          ],
          "className": "p-toggle-option",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "LGBTQIA+",
            },
            "ref": null,
            "rendered": "LGBTQIA+",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "enabled": true,
              "name": "lgbtq",
              "on": false,
              "onClick": [Function],
              "onDisabledClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Female-Identifying
            </p>,
            <Switch
              className=""
              enabled={true}
              name="female"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />,
          ],
          "className": "p-toggle-option",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Female-Identifying",
            },
            "ref": null,
            "rendered": "Female-Identifying",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "enabled": true,
              "name": "female",
              "on": false,
              "onClick": [Function],
              "onDisabledClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Male-Identifying
            </p>,
            <Switch
              className=""
              enabled={true}
              name="male"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />,
          ],
          "className": "p-toggle-option",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Male-Identifying",
            },
            "ref": null,
            "rendered": "Male-Identifying",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "enabled": true,
              "name": "male",
              "on": false,
              "onClick": [Function],
              "onDisabledClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Front-End
            </p>,
            <Switch
              className=""
              enabled={true}
              name="frontEnd"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />,
          ],
          "className": "p-toggle-option",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Front-End",
            },
            "ref": null,
            "rendered": "Front-End",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "enabled": true,
              "name": "frontEnd",
              "on": false,
              "onClick": [Function],
              "onDisabledClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Back-End
            </p>,
            <Switch
              className=""
              enabled={true}
              name="backEnd"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />,
          ],
          "className": "p-toggle-option",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Back-End",
            },
            "ref": null,
            "rendered": "Back-End",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "enabled": true,
              "name": "backEnd",
              "on": false,
              "onClick": [Function],
              "onDisabledClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="p-toggle-option"
          >
            <p>
              Veteran
            </p>
            <Switch
              className=""
              enabled={true}
              name="veteran"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />
          </div>,
          <div
            className="p-toggle-option"
          >
            <p>
              Parent
            </p>
            <Switch
              className=""
              enabled={true}
              name="parent"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />
          </div>,
          <div
            className="p-toggle-option"
          >
            <p>
              LGBTQIA+
            </p>
            <Switch
              className=""
              enabled={true}
              name="lgbtq"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />
          </div>,
          <div
            className="p-toggle-option"
          >
            <p>
              Female-Identifying
            </p>
            <Switch
              className=""
              enabled={true}
              name="female"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />
          </div>,
          <div
            className="p-toggle-option"
          >
            <p>
              Male-Identifying
            </p>
            <Switch
              className=""
              enabled={true}
              name="male"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />
          </div>,
          <div
            className="p-toggle-option"
          >
            <p>
              Front-End
            </p>
            <Switch
              className=""
              enabled={true}
              name="frontEnd"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />
          </div>,
          <div
            className="p-toggle-option"
          >
            <p>
              Back-End
            </p>
            <Switch
              className=""
              enabled={true}
              name="backEnd"
              on={false}
              onClick={[Function]}
              onDisabledClick={[Function]}
            />
          </div>,
        ],
        "className": "hide",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Veteran
              </p>,
              <Switch
                className=""
                enabled={true}
                name="veteran"
                on={false}
                onClick={[Function]}
                onDisabledClick={[Function]}
              />,
            ],
            "className": "p-toggle-option",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Veteran",
              },
              "ref": null,
              "rendered": "Veteran",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "enabled": true,
                "name": "veteran",
                "on": false,
                "onClick": [Function],
                "onDisabledClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Parent
              </p>,
              <Switch
                className=""
                enabled={true}
                name="parent"
                on={false}
                onClick={[Function]}
                onDisabledClick={[Function]}
              />,
            ],
            "className": "p-toggle-option",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Parent",
              },
              "ref": null,
              "rendered": "Parent",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "enabled": true,
                "name": "parent",
                "on": false,
                "onClick": [Function],
                "onDisabledClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                LGBTQIA+
              </p>,
              <Switch
                className=""
                enabled={true}
                name="lgbtq"
                on={false}
                onClick={[Function]}
                onDisabledClick={[Function]}
              />,
            ],
            "className": "p-toggle-option",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "LGBTQIA+",
              },
              "ref": null,
              "rendered": "LGBTQIA+",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "enabled": true,
                "name": "lgbtq",
                "on": false,
                "onClick": [Function],
                "onDisabledClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Female-Identifying
              </p>,
              <Switch
                className=""
                enabled={true}
                name="female"
                on={false}
                onClick={[Function]}
                onDisabledClick={[Function]}
              />,
            ],
            "className": "p-toggle-option",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Female-Identifying",
              },
              "ref": null,
              "rendered": "Female-Identifying",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "enabled": true,
                "name": "female",
                "on": false,
                "onClick": [Function],
                "onDisabledClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Male-Identifying
              </p>,
              <Switch
                className=""
                enabled={true}
                name="male"
                on={false}
                onClick={[Function]}
                onDisabledClick={[Function]}
              />,
            ],
            "className": "p-toggle-option",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Male-Identifying",
              },
              "ref": null,
              "rendered": "Male-Identifying",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "enabled": true,
                "name": "male",
                "on": false,
                "onClick": [Function],
                "onDisabledClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Front-End
              </p>,
              <Switch
                className=""
                enabled={true}
                name="frontEnd"
                on={false}
                onClick={[Function]}
                onDisabledClick={[Function]}
              />,
            ],
            "className": "p-toggle-option",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Front-End",
              },
              "ref": null,
              "rendered": "Front-End",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "enabled": true,
                "name": "frontEnd",
                "on": false,
                "onClick": [Function],
                "onDisabledClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Back-End
              </p>,
              <Switch
                className=""
                enabled={true}
                name="backEnd"
                on={false}
                onClick={[Function]}
                onDisabledClick={[Function]}
              />,
            ],
            "className": "p-toggle-option",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Back-End",
              },
              "ref": null,
              "rendered": "Back-End",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "enabled": true,
                "name": "backEnd",
                "on": false,
                "onClick": [Function],
                "onDisabledClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
